Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,25 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    size_t buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        if (size != sizeof(size_t))
+            return EINVAL;
+        buffer_size = m_for_server->immediately_readable();
```

Comment:
This appears to return the amount of data that is currently buffered. `SO_SNDBUF` and `SO_RCVBUF` are supposed to return the total buffer size.